stan/math/prim/fun/mdivide_left_tri.hpp

```diff
@@ -89,6 +96,10 @@ inline Eigen::Matrix<double, R1, C2> mdivide_left_tri(
     const Eigen::Matrix<double, R2, C2> &b) {
   check_square("mdivide_left_tri", "A", A);
   check_multiplicable("mdivide_left_tri", "A", A, "b", b);
+  if (A.rows() == 0) {
+    return Eigen::MatrixXd(0, b.cols());
```

Comment:
This should be `return {0, b.cols()}` so that the object constructed matches the return type, which is `Eigen::Matrix<double, R1, C2>` here.  I don't see any requirement that `R1 = C2 = -1` as implied by `Eigen::MatrixXd`.  

It'll work as written because Eigen ignores the size templates in assignment, but it's confusing that the types don't match.